src/main/java/org/traccar/database/NotificationManager.java

```diff
@@ -92,9 +92,12 @@ public void updateEvent(Event event, Position position) {
                     if (getById(notificationId).getType().equals(event.getType())) {
                         boolean filter = false;
                         if (event.getType().equals(Event.TYPE_ALARM)) {
-                            String alarms = notification.getString("alarms");
-                            if (alarms == null || !alarms.contains(event.getString(Position.KEY_ALARM))) {
+                            String alarmsAttribute = notification.getString("alarms");
+                            if (alarmsAttribute == null) {
                                 filter = true;
+                            } else {
+                                List<String> alarms = Arrays.asList(alarmsAttribute.split(","));
```

Comment:
One last thought. Should we use `Set` instead of `Array`?

Reply:
Good question. Since it is redundant in this case, I would prefer the less verbose option here for simplicity. For java 9+ we can use something like `Set<String> alarms = Set.of(alarmsAttribute.split(","));`. What do you think?

Reply:
I think we can keep as is for now.